### PR TITLE
[release/3.1] Remove old unused Docker files

### PR DESCRIPTION
**Pull request by dougbu**
- in part, this is a `cherry-pick` of c5c8f4385ee8
- also cleans up the old SystemdActivation files

### CODE REVIEW

File: src/Servers/Kestrel/test/SystemdActivation/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet-nightly:2.0-runtime-deps
+FROM mcr.microsoft.com/dotnet/runtime-deps:2.1
```
Comment (dougbu) [inline src/Servers/Kestrel/test/SystemdActivation/Dockerfile:1]: Should we instead be using the `3.1` tag in this branch❔

File: src/Servers/Kestrel/samples/Http2SampleApp/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/aspnetcore:2.0.0-stretch
+FROM mcr.microsoft.com/dotnet/aspnet:2.1-stretch-slim
```
Comment (dougbu) [inline src/Servers/Kestrel/samples/Http2SampleApp/Dockerfile:1]: Other alternatives:

1. use `mcr.microsoft.com/dotnet/aspnet:3.1-buster-slim` tag (which is Docker 10)
2. backport c5c8f4385ee8

**Review by wtgodbe — APPROVED** [review]
I'm cool with this as long as it works (which it appears to)

**Comment by dougbu** [comment]
Reset to just delete the relevant files

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**